### DISCUSSION TITLE
[4.x]: Payment source is not associated to the order

### DISCUSSION THREAD

**Discussion by goraxan** [craftcms/commerce#3102] [General]
### Description
When paying, the payment source is saved correctly when using the `savePaymentSource` in the `paymentsources` table but the `paymentSourceId` column from the orders table is not populated, so I'm not able to know which payment source is used to which order.

This is the request to commerce/payments/pay controller
![image](https://user-images.githubusercontent.com/8337784/221560679-53d2643e-6e84-47b0-8e98-5ecc9ab21a9a.png)

The paymentsource correctly created
![image](https://user-images.githubusercontent.com/8337784/221560896-d6368c10-f9b1-4859-bd47-505e0f562fbb.png)

The order not being updated correctly (both gatewayId and paymentSourceId are NULL)
![image](https://user-images.githubusercontent.com/8337784/221561077-6f0f4eb6-b98d-41ac-85a7-1e73ed4360d6.png)

### Craft CMS version

4.3.6.1

### Craft Commerce version

4.2.4

### PHP version

8.0.26

### Installed plugins and versions

-

**Comment by pdaleramirez** [comment]
@goraxan That is the expected value for that paymentSourceId column. The attribute on the order is used just during the payment process.

More details: https://github.com/craftcms/commerce/blob/develop/src/elements/Order.php#L1611-L1614

We’ll be looking into saving the paymentSourceId in the transaction table In future releases.

**Comment by goraxan** [comment]
@pdaleramirez oh, I see. Associating it to the transaction instead of the order makes sense, thank you!
